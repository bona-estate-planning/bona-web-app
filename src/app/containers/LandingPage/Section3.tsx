/**
 *
 * Section3
 *
 */
import { useMediaQuery, useTheme } from '@material-ui/core';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import Container from '@material-ui/core/Container';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React, { memo } from 'react';

interface Props {}

const useStyles = makeStyles(theme => {
  return {
    section: (isMd: boolean) => ({
      padding: theme.spacing(7, 0, isMd ? 13 : 3.5),
      textAlign: 'center',
    }),
    headline: {
      fontWeight: theme.typography.fontWeightBold,
      fontSize: theme.spacing(3.5),
      marginBottom: theme.spacing(3),
    },
    subtitle: {
      fontSize: theme.spacing(2.25),
      marginBottom: theme.spacing(0.5),
    },
    plans: {
      margin: theme.spacing(0),
    },
    plan: {
      display: 'inline-block',
      width: 278,
      margin: theme.spacing(3.5),
      padding: theme.spacing(4.5, 1.5),
      backgroundColor: theme.palette.grey[100],
    },
    planTitle: {
      fontSize: theme.spacing(2.2),
      fontWeight: theme.typography.fontWeightBold,
      marginBottom: theme.spacing(2),
    },
    planSubtitle: {
      marginBottom: theme.spacing(3),
    },
    planPrice: {
      fontSize: theme.spacing(5),
      fontWeight: theme.typography.fontWeightBold,
      marginBottom: theme.spacing(2),
    },
    planButton: {
      borderRadius: 99999,
      padding: theme.spacing(1, 6),
      textTransform: 'none',
      fontWeight: theme.typography.fontWeightBold,
    },
  };
});

export const Section3 = memo((props: Props) => {
  const theme = useTheme();
  const isMd = useMediaQuery(theme.breakpoints.up('md'));
  const classes = useStyles(isMd);

  const plans = [
    {
      title: 'Guardianship',
      subtitle: 'Ensure your family is cared for with all life throws at you',
      price: 50,
      to: '/',
    },
    {
      title: 'Advanced Directive',
      subtitle: 'Sort out complex medical treatment details',
      price: 50,
      to: '/',
    },
    {
      title: 'HIPAA Authorization',
      subtitle: 'Decide the access to your medical information',
      price: 50,
      to: '/',
    },
    {
      title: 'Trust & Will',
      subtitle: 'Protect and transfer your most important assets',
      price: 50,
      to: '/',
    },
  ];

  return (
    <Box className={classes.section}>
      <Container>
        <Typography component="h3" className={classes.headline}>
          We got you covered
        </Typography>
        <Typography component="p" className={classes.subtitle}>
          Whatever your trust concerns are we have you covered. Let us navigate
          the complex waters of estate planning for you.
        </Typography>
      </Container>
      <Box className={classes.plans}>
        {plans.map(({ title, subtitle, price, to }, i) => (
          <Paper key={`plan-${i}`} className={classes.plan} elevation={3}>
            <Typography component="p" className={classes.planTitle}>
              {title}
            </Typography>
            <Typography
              component="p"
              variant="subtitle1"
              className={classes.planSubtitle}
            >
              {subtitle}
            </Typography>
            <Typography component="p" className={classes.planPrice}>
              ${price}
            </Typography>
            <Button
              variant="contained"
              color="primary"
              className={classes.planButton}
            >
              Learn More
            </Button>
          </Paper>
        ))}
      </Box>
    </Box>
  );
});
