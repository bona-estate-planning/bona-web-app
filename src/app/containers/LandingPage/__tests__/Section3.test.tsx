import React from 'react';
import { render } from '@testing-library/react';
import { Section3 } from '../Section3';

describe('<Section3  />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Section3 />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
