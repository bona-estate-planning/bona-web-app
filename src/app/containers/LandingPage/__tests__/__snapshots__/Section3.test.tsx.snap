// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section3  /> should match snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-11 makeStyles-section-1 makeStyles-section-10"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h3
      class="MuiTypography-root makeStyles-headline-2 MuiTypography-body1"
    >
      We got you covered
    </h3>
    <p
      class="MuiTypography-root makeStyles-subtitle-3 MuiTypography-body1"
    >
      Whatever your trust concerns are we have you covered. Let us navigate the complex waters of estate planning for you.
    </p>
  </div>
  <div
    class="MuiBox-root MuiBox-root-12 makeStyles-plans-4"
  >
    <div
      class="MuiPaper-root makeStyles-plan-5 MuiPaper-elevation3 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root makeStyles-planTitle-6 MuiTypography-body1"
      >
        Guardianship
      </p>
      <p
        class="MuiTypography-root makeStyles-planSubtitle-7 MuiTypography-subtitle1"
      >
        Ensure your family is cared for with all life throws at you
      </p>
      <p
        class="MuiTypography-root makeStyles-planPrice-8 MuiTypography-body1"
      >
        $
        50
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-planButton-9 MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Learn More
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root makeStyles-plan-5 MuiPaper-elevation3 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root makeStyles-planTitle-6 MuiTypography-body1"
      >
        Advanced Directive
      </p>
      <p
        class="MuiTypography-root makeStyles-planSubtitle-7 MuiTypography-subtitle1"
      >
        Sort out complex medical treatment details
      </p>
      <p
        class="MuiTypography-root makeStyles-planPrice-8 MuiTypography-body1"
      >
        $
        50
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-planButton-9 MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Learn More
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root makeStyles-plan-5 MuiPaper-elevation3 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root makeStyles-planTitle-6 MuiTypography-body1"
      >
        HIPAA Authorization
      </p>
      <p
        class="MuiTypography-root makeStyles-planSubtitle-7 MuiTypography-subtitle1"
      >
        Decide the access to your medical information
      </p>
      <p
        class="MuiTypography-root makeStyles-planPrice-8 MuiTypography-body1"
      >
        $
        50
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-planButton-9 MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Learn More
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root makeStyles-plan-5 MuiPaper-elevation3 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root makeStyles-planTitle-6 MuiTypography-body1"
      >
        Trust & Will
      </p>
      <p
        class="MuiTypography-root makeStyles-planSubtitle-7 MuiTypography-subtitle1"
      >
        Protect and transfer your most important assets
      </p>
      <p
        class="MuiTypography-root makeStyles-planPrice-8 MuiTypography-body1"
      >
        $
        50
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-planButton-9 MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Learn More
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
